src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -0,0 +1,403 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+using System.Threading;
+using System.Runtime.InteropServices;
+
+using CFStringRef = System.IntPtr;
+using CFArrayRef = System.IntPtr;
+using FSEventStreamRef = System.IntPtr;
+using size_t = System.IntPtr;
+using FSEventStreamEventId = System.UInt64;
+using CFTimeInterval = System.Double;
+using CFRunLoopRef = System.IntPtr;
+
+namespace System.IO
+{
+    // OSX implementation of FileSystemWatcher's PAL.  The main partial definition of FileSystemWatcher expects:
+    // - StartRaisingEvents
+    // - StopRaisingEvents
+    // - FinalizeDispose
+
+    public partial class FileSystemWatcher
+    {
+        // OS X has a case-sensitive File System
+        internal const bool CaseSensitive = true;
```

Comment:
After you forked the file from the Linux version, I tweaked the organization of it to put everything expected by the PAL above a comment and everything else below a comment.  It'd be good to do the same here.  Take a look at the current state of the Linux file for reference.